### ISSUE TITLE
th-has-data-cells fails on empty tables

### ISSUE THREAD

**Issue by just-boris** [dequelabs/axe-core#2508]
I ran my pages using axe checker and it seems to fail on empty tables:

```html
<table>
  <thead>
    <tr>
      <th>Product</th>
      <th>Amount</th>
      <th>Price</th>
    </tr>
  </tbody>
  <tbody>
    <tr>
      <td colspan="3">No available products to display</td>
    </tr>
  </thead>
</table>
```

Live demo https://jsbin.com/pewali

Expectation: axe handles this table and does not report `incomplete` checks

Actual: axe stumbles on this table and reports `incomplete` check

Motivation: Incomplete checks still require attention and should not give false-negatives

<pre><code>
axe-core version: 3.5.5
chrome-extension: 4.5.3
</code></pre>

**Comment by WilcoFiers** [comment]
Yup, you seem to have found a bug! Thank you for reporting. We'll definitely fix this.

**Comment by straker** [comment]
Note on the bug: It seems our `getHeaders` and `getCellPosition` functions do not take into account row/col-span. So when we look at the cells for the 2nd and 3rd headers and ask what the header of those cells are, it only returns the first header and not the other 2, so we fail saying that the cell points to a different header.

**Comment by padmavemulapati** [comment]
Verified with the latest axe-core(dated 22/10/2020) - 4.0.2 version
No `incomplete` results found for the table' rowspan and colspan 
![image](https://user-images.githubusercontent.com/47409579/96870264-432f0a80-148e-11eb-89a9-0cc0808a7148.png)

Also verified on 3.5.4 version where it is showing up `incomplete` results. (So fix is verified on 4.0.2 version)
![Uploading image.png…]()
